src/com/cft/contactmerge/NameMatchLogic.java

```diff
@@ -0,0 +1,129 @@
+/* Created by James Page on May 8th, 2018
+ * Contains the low level business logic for matching naming properties
+ */
+
+package com.cft.contactmerge;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+public abstract class NameMatchLogic {
+    /***********************************************************************
+     ****************************** Variables ******************************
+     ***********************************************************************/
+    private final static int hyphenCharLimitYesVsMaybe = 3;
+    private final static String[] namePunctuations = {".", ",", "'", "-"};
+    protected final static String[][] alternateFirstNames = // List of names to match nicknames of the same person
+            {
+                    {"Al", "Allen"},
+                    {"Al", "Alan"},
+                    {"Bob", "Robert"}
+            };
+
+    /***********************************************************************
+     ********************* Names Compare Logic Methods *********************
+     ***********************************************************************/
+    protected static boolean isNamesExactMatch(ArrayList<String> nameOne, ArrayList<String> nameTwo){
+        return nameOne.equals(nameTwo);
+    }
+
+    protected static boolean doNamesContainMultipleParts(ArrayList<String> nameOne, ArrayList<String> nameTwo){
+        return nameOne.size() > 1 || nameTwo.size() > 1;
+    }
+
+    protected static boolean doNamesContainMultipleParts(ArrayList<String> nameOne, ArrayList<String> nameTwo, ArrayList<String> nameThree, ArrayList<String> nameFour){
+        return nameOne.size() > 1 || nameTwo.size() > 1 || nameThree.size() > 1 || nameFour.size() > 1;
+    }
+
+    protected static boolean doNamesContainSingleCharacterOnly(ArrayList<String> nameOne, ArrayList<String> nameTwo){
+        return nameOne.get(0).length() == 1 || nameTwo.get(0).length() == 1;
+    }
+
+    protected static AnswerType getHyphinatedNamesMatchResults(ArrayList<String> hNameOne, ArrayList<String> hNameTwo){
+        if (getNumOfCharsInName(hNameOne) > hyphenCharLimitYesVsMaybe && getNumOfCharsInName(hNameTwo) > hyphenCharLimitYesVsMaybe){
```

Comment:
The title suggested that we are checking if the names match but the code is just checking if the size of both names is greater than the hyphen limit. Either the method name is misleading or your are missing a check for equality.

Reply:
nice catch. I will rename this method.

Reply:
I'm changing it to "getYesOrMaybeHyphinatedNameMatch". This means the names being passed have already decided a match was found, but now based on the 3char rule (our current default number), it will return if it's a yes or maybe. If you can think of a better name let me know and I will change it again.